File: backend/app/schemas/category.py
from pydantic import BaseModel
from typing import Optional


class CategoryBase(BaseModel):
    name_ar: str
    name_en: str
    icon_url: Optional[str] = None
    sort_order: int = 0


class CategoryCreate(CategoryBase):
    pass


class CategoryResponse(CategoryBase):
    id: int

    class Config:
        from_attributes = True  # للتوافق مع كائنات SQLAlchemy مباشرة

File: syria_offers_app/lib/models/category.dart
class Category {
  final int id;
  final String nameAr;
  final String nameEn;
  final String? iconUrl;
  final int sortOrder;

  Category({
    required this.id,
    required this.nameAr,
    required this.nameEn,
    this.iconUrl,
    required this.sortOrder,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      nameAr: json['name_ar'],
      nameEn: json['name_en'],
      iconUrl: json['icon_url'],
      sortOrder: json['sort_order'],
    );
  }
}